### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by brandy223**
*Few corrections and changes made for the new Academic year 2023-2024.*

__List of changes :__
- ➕ Secretary Role
- ➕ Bachelor Cyber
- 🔁 "Big Data" > "Développement Logiciel"
- 🔁 Form : updated lv2 options
- ✅ Typos

### CODE REVIEW

**Review by tom-bourjala — DISMISSED** [review]
Some welcome updates, the user.js "roleCount" should get a refactor at some point...
Looks fine to me, no issues that come to my mind. Sigma should create the roles at reboot. "Big Data" role may need a manual delete.

File: bot/requests/modules/planning/core.js [hunk 1]
```diff
@@ -28,7 +28,7 @@ async function coreProcess(author, dates = [], search = null, target = null){
         ical = await getUserICS(namedotsurname);
     }catch(err){
         if(global.debug) console.log(err);
-        return "Une erreur s'est prduite à la récupération de l'emplois-du-temps. Veuillez réessayer.\n (*" + err + "*)";
+        return "Une erreur s'est produite à la récupération de l'emplois-du-temps. Veuillez réessayer.\n (*" + err + "*)";
```
Comment (Sigeth) [inline bot/requests/modules/planning/core.js]: l'emplois-du-temps => l'emploi du temps

**Review by Sigeth — COMMENTED** [review]
Should be good, just change the line I mentioned to remove this french error

**Comment by brandy223** [comment]
> Some welcome updates, the user.js "roleCount" should get a refactor at some point... Looks fine to me, no issues that come to my mind. Sigma should create the roles at reboot. "Big Data" role may need a manual delete.

Concerning roleCount, I planned on possibly reworking the entire association module, but that will be one of the main things to change. And yes, that could be great if Sigma generates all classes roles at boot, I'll add it to the to-do list.

File: bot/requests/modules/planning/core.js (hunk 1 above)
Comment (brandy223) [reply bot/requests/modules/planning/core.js]: This is fixed [here](https://github.com/ISEN-Promo-Pole-developpement-Discord/Sigma-VI/pull/46/commits/3a6c981554a3642d537c1f4e467639ce0ef3288e).

**Review by Sigeth — APPROVED** [review]
Good now 👍

**Pull request merged**